protobuf-maven-plugin/src/main/java/io/github/ascopes/protobufmavenplugin/generation/SourceCodeGenerator.java

```diff
@@ -118,11 +118,11 @@ public boolean generate(GenerationRequest request) throws ResolutionException, I
 
     var argLineBuilder = new ArgLineBuilder(protocPath)
         .fatalWarnings(request.isFatalWarnings())
-        .importPaths(importPaths
-            .stream()
-            .map(ProtoFileListing::getProtoFilesRoot)
-            .collect(Collectors.toCollection(LinkedHashSet::new)))
-        .importPaths(request.getSourceRoots());
+        .importPaths(
```

Comment:
think we want to keep including the source roots so they are on the import path as well. Shall see what the CI build does though first.

The assumption should be that the linked hash set will be deduplicating, but I don't think that is working by the looks.

Let's see what the build does, as it may be totally unneeded like you have observed!

Reply:
Oh, looks like it is fine! Good spot. Thanks for this. Will merge!

Reply:
yeah, the previous logic already included the source roots and filtered it for dupes, so I figured it was okay to remove. Thanks!